lib/src/components/AbstractDialog/AbstractDialog.tsx

```diff
@@ -0,0 +1,189 @@
+ */
+
+// propTypes static analysis doesn't work with the way this component is written
+/* eslint-disable react/prop-types */
+const AbstractDialog = forwardRef<HTMLDialogElement, Props>((props, ref) => {
+  const {
+    className,
+    children,
+    onCancel,
+    useNativeCancelOnEscape,
+    role,
+    isModal,
+    ...attrs
+  } = props;
+
+  const dialogRef = useRef<HTMLDialogElement>(null);
+
+  // The dialogRef value needs to get passed down in a context. But athe context needs to know when the ref
+  // value has updated, and refs aren't tracked like state values. So we have to copy the ref value into a state
+  // value in order for it to be tracked.
+  const [dialogRefState, setDialogRefState] = useState<HTMLDialogElement | null>(null);
+
+  const mergedRef = useMergedRef(dialogRef, ref);
+
+  function dialogKeydownListener(evt: KeyboardEvent<HTMLDialogElement>) {
+    if (evt.key === 'Escape' || evt.key === 'Esc') {
+      // HACK for backwards compatibility: it is known that some downstream uses of NxModal do not provide
+      // the onCancel/onCancel handler despite it being required. It is also known that these downstream uses
+      // have some logic which globally manages ESC handlers for a variety of things including modals. To keep that
+      // working, only stopPropagation if the onCancel callback is defined. If it isn't defined, tenuously assume
+      // that the ESC handling is implemented externall and do nothing here.
+      if (onCancel) {
+        // prevent visibility of the keydown outside of the dialog, so that global ESC listeners on the
+        // document don't pick it up
+        evt.stopPropagation();
+
+        // prevent visibility to manually-registered native event listeners on the document too.
+        // NOTE: this only works on listeners added after this one, which is believed to include any
+        // registered in useEffect calls on components rendered simultaneously with the dialog
+        evt.nativeEvent.stopImmediatePropagation();
+
+        if (!evt.defaultPrevented) {
+          if (!useNativeCancelOnEscape) {
+            evt.preventDefault();
+            onCancel(createCancelEvent());
+          }
+          else if (!hasNativeModalSupport) {
+            // emulate cancel-on-esc behavior in browsers which don't do it natively
+            onCancel(createCancelEvent());
+          }
+        }
+      }
+    }
+  }
+
+  useEffect(function() {
+    /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+    const el = dialogRef.current!,
+        // HTML <dialog> elements are supposed to remember what element was focused before they were opened,
+        // and restore focus to that element when they are closed. No browsers appear to implement this currently,
+        // so we do it ourselves
+        previouslyFocusedEl = document.activeElement;
+
+    setDialogRefState(el);
+
+    /*
+     * This will cause the document to become "blocked by the modal dialog"
```

Comment:
This comment only applies to the modal case, and so should be moved inside that part of the condition

Reply:
> This comment only applies to the modal case, and so should be moved inside that part of the condition

Fixed here: 45e2afe18db26baf63004bbf05b75f863d5a77e8